core/message_pool.go

```diff
@@ -7,25 +7,50 @@ import (
 	"github.com/ipfs/go-cid"
 	"github.com/pkg/errors"
 
+	"github.com/filecoin-project/go-filecoin/actor"
 	"github.com/filecoin-project/go-filecoin/address"
 	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/consensus"
 	"github.com/filecoin-project/go-filecoin/metrics"
+	"github.com/filecoin-project/go-filecoin/state"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
 var mpSize = metrics.NewInt64Gauge("message_pool_size", "The size of the message pool")
 
+// MaxMessagePoolSize is the maximum number of pending messages will will allow in the message pool at any time
+const MaxMessagePoolSize = 10000
+
+// MaxNonceGap is the maximum nonce of a message past the last received on chain
+const MaxNonceGap = 100
+
 // MessageTimeOut is the number of tipsets we should receive before timing out messages
 const MessageTimeOut = 6
 
+var (
+	ErrMessagePoolFull  = errors.New("message pool is full")
+	ErrNonceGapExceeded = errors.New("message nonce is too much greater than highest nonce on chain")
+	ErrDuplicateNonce   = errors.New("different message with same from address and nonce is already in pool")
+)
+
 type timedmessage struct {
 	message *types.SignedMessage
 	addedAt uint64
 }
 
-// BlockTimer defines a interface to a struct that can give the current block height.
-type BlockTimer interface {
+// MessagePoolAPI defines a interface api resources the message pool needs.
```

Comment:
english, please. also this isn't really the message pool api, it's the message pool's dependencies.